Fixes drawing outside of plot by adding axis var

xksteven opened the pull request:
Fixes drawing outside of plot by adding axis variable to roipoly.  The if event.inaxes only checks if it is not None which will be false inside of buttons.

xksteven committed:
Fixes drawing outside of plot by adding axis var

Fixes drawing outside of plot by adding axis variable to roipoly.  The if event.inaxes only checks if it is not None which will be false inside of buttons.

jdoepfert commented on roipoly/roipoly.py:
```diff
@@ -22,7 +22,7 @@ def deprecation(message):
 class RoiPoly:
 
-    def __init__(self, fig=None, color='b',
+    def __init__(self, fig=None, ax=None, color='b',
```
could you add the new argument to the docstring as well?

xksteven replied:
Just added.  The change slipped my notice.

jdoepfert commented:
Thx for ur contribution.
When I run `examples/multi_roi_example.py`, start drawing a new ROI, and then, while drawing, hover over one of the buttons, the current ROI-endpoint will move to the upper left of the image axes. Any idea why?
![image](https://user-images.githubusercontent.com/7914340/65813474-21ede300-e1d6-11e9-9c83-28eabd66f90d.png)
Any idea how to fix this?

xksteven committed:
Adds ax to the parameter documentation

Updates the parameter documentation to add reference to the added axes variable.

xksteven commented:
When hovering over the button, the button's location is something like (0.8,0.2) which translates to a point near the very top left of the image.  Because the button has it's own Axes the mouse move callback will attempt to draw the line to the new (x,y) position as it enters the button area.

The line moving to the top left behavior should be fixed with this pull request though.  I just tested it to make sure it was the case on my machine.

jdoepfert committed:
fix docstring

jdoepfert commented:
thanks for ur contribution! will be merged after tests.
fixes #19
